addons/web/static/tests/views/list_tests.js

```diff
@@ -428,6 +428,286 @@ QUnit.module('Views', {
         list.destroy();
     });
 
+    QUnit.test('list view: action button in controlPanel basic rendering', async function (assert) {
+        assert.expect(12);
+
+        // LPE FIXME
+        // mockServer doesn't do the right thing (ie python like) for modifiers
+        const list = await createView({
+            View: ListView,
+            model: 'foo',
+            data: this.data,
+            arch: `
+                <tree>
+                    <header>
+                         <button name="x" type="object" class="plaf" string="plaf"/>
+                         <button name="y" type="object" class="plouf" string="plouf" modifiers='{"invisible": true}'/>
+                    </header>
+                    <field name="foo" />
+                </tree>`,
+        });
+        let cpButtons = cpHelpers.getButtons(list);
+        assert.containsNone(
+            cpButtons[0],
+            'button[name="x"]'
+        );
+        assert.containsNone(
+            cpButtons[0],
+            '.o_list_selection_box'
+        );
+        assert.containsNone(
+            cpButtons[0],
+            'button[name="y"]'
+        );
+
+        await testUtils.dom.click(
+            list.el.querySelector('.o_data_row .o_list_record_selector input[type="checkbox"]')
+        );
+        cpButtons = cpHelpers.getButtons(list);
+        assert.containsOnce(
+            cpButtons[0],
+            'button[name="x"]'
+        );
+        assert.hasClass(
+            cpButtons[0].querySelector('button[name="x"]'),
+            'btn'
+        );
+        assert.hasClass(
+            cpButtons[0].querySelector('button[name="x"]'),
+            'btn-secondary'
```

Comment:
I think hasClass works with multiple classNames, so you can merge the last two assertions